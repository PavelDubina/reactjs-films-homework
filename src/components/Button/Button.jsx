import React, { Children } from 'react'
import PropTypes from 'prop-types'

import './Button.scss'

const Button = ({ styles, onViewInfo, children }) => {
  return (
    <button className={styles} onClick={onViewInfo}>
      {children}
    </button>
  )
}

export default Button
