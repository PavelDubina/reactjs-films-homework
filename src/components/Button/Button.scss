button {
  width: 110px;
  height: 35px;
  padding: 5px;
  margin: 0 10px;
  border: none;
  border-radius: 5px;
  outline: none;
  font-size: 16px;
  cursor: pointer;

  &:active {
    opacity: 0.95;
  }
}
