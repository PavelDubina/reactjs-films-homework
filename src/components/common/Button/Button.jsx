import React from 'react'
import PropTypes from 'prop-types'
import className from 'classnames'
import styles from './Button.scss'

export const Button = ({ type = 'primary', onClick, children }) => {
  const classList = className(styles.button, {
    [styles.primary]: type === 'primary',
    [styles.secondary]: type === 'secondary',
    [styles.info]: type === 'info',
  })
  return (
    <button className={classList} onClick={onClick}>
      {children}
    </button>
  )
}

Button.propTypes = {
  type: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
}
