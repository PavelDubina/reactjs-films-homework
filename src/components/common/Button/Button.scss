.button {
  width: 110px;
  height: 35px;
  padding: 5px;
  margin: 0 10px;
  border: none;
  border-radius: 5px;
  outline: none;
  font-size: 16px;
  cursor: pointer;

  &:active {
    opacity: 0.95;
  }

  @media (max-width: 599px) {
    height: 32px;
    width: 100px;
    margin: 0 5px;
  }
}

.primary {
  background-color: rgb(24, 183, 236);
  color: #fff;
}

.secondary {
  background-color: inherit;
  border: 1px solid #fff;
  color: #fff;
}

.info {
  background-color: #f0f0f0;
  color: rgb(24, 183, 236);
}
