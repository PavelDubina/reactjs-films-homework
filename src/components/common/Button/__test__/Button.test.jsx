/*
 * Copyright Â© 2020 EPAM Systems, Inc. All Rights Reserved. All information contained herein is, and remains the
 * property of EPAM Systems, Inc. and/or its suppliers and is protected by international intellectual
 * property law. Dissemination of this information or reproduction of this material is strictly forbidden,
 * unless prior written permission is obtained from EPAM Systems, Inc
 */

import React from 'react'
import { create } from 'react-test-renderer'
import { Button } from '../Button'

describe('Button', () => {
  it('Should render and match snapshot', () => {
    const mockFn = jest.fn()
    const component = create(<Button onCLick={mockFn}>Button</Button>)
    component.root.props.onClick()
    expect(component.toJSON()).toMatchSnapshot()
    expect(mockFn).toBeCalledTimes(1)
  })
})
