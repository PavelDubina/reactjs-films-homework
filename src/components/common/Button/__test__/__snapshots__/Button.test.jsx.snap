// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render and match snapshot 1`] = `
<button
  className="button primary"
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  Button
</button>
`;
